File: programming-concepts/iteration/java/nested_iteration_guess_numbers.java
/*
Raspberry Pi Foundation
Developed to be used alongside Isaac Computer Science,
part of the National Centre for Computing Education
Usage licensed under CC BY-SA 4

Note: This file is designed to be copied out and compiled on your machine.
In order for the program to compile properly in some IDEs, you need to ensure that the
class filename is the same as the class name with the main method in it.

To run this file you need to:
1. Copy the contents
2. Create a Java project in the IDE of your choice (Eclipse, for example)
3. Create a new Java Class that uses the same name as the class containing the main method in this program
4. Paste the contents into the new Java Class
5. Save and run the program
*/

import java.util.Scanner;

class Iteration
{
    // The main method is the entry point for all Java programs
    public static void main(String[] args)
    {
        // Use Scanner for reading user input from the console
        Scanner scanner = new Scanner(System.in);

        int[] numbersToGuess = { 1, 4, 8, 3, 10 };

        System.out.println("Guess my numbers, each number is between 1 and 10");

        for (int number : numbersToGuess) {
            System.out.println("Enter a number to guess: ");
            String userInput = scanner.nextLine();
            int guess = Integer.parseInt(userInput);

            while (guess != number) {
                System.out.println("Incorrect number. Try again: ");
                userInput = scanner.nextLine();
                guess = Integer.parseInt(userInput);
            }

            System.out.println("Well done, next number to guess");
        }
    }


}

File: programming-concepts/iteration/python/nested_iteration_guess_numbers.py
# Raspberry Pi Foundation
# Developed as part of Ada Computer Science
# 
# Usage licensed under CC BY-NC-SA 4.0

def main():
    numbers_to_guess = [1,4,8,3,10]

    print("Guess my numbers, each number is between 1 and 10")

    # Repeat for each number in the list
    for number in numbers_to_guess:
        user_input = input("Enter a number to guess: ")
        guess = int(user_input)

        # Repeat while the user has not guessed the correct number
        while guess != number:
            user_input = input("Incorrect number. Try again: ")
            guess = int(user_input)

        print("Well done, next number to guess")


# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == '__main__':
    main()
